mkosi/__init__.py

```diff
@@ -2439,6 +2440,12 @@ def finalize_tools(args: MkosiArgs, presets: Sequence[MkosiConfig]) -> Sequence[
         release = p.tools_tree_release or distribution.default_release()
         mirror = p.tools_tree_mirror or (p.mirror if p.mirror and p.distribution == distribution else None)
 
+        if p.cache_dir:
+            same = p.distribution == distribution and p.release == release and p.architecture == Architecture.native()
+            cache = p.cache_dir if same else p.cache_dir / "tools"
+        else:
+            cache = None
```

Comment:
It's a bit more vertical space but I think it takes a bit less mental gymnastics by not taking the predicate inside out
```suggestion
        if p.cache_dir:
            if p.distribution == distribution and p.release == release and p.architecture == Architecture.native():
                cache = p.cache_dir
            else:
                cache = p.cache_dir / "tools"
        else:
            cache = None
```